getconf/base.py

```diff
@@ -275,6 +276,34 @@ def getfloat(self, key, default=0.0, doc=''):
             logger.exception("Unable to cast %s as float for the key %s.", repr(value), key)
             raise
 
+    def gettimespan(self, key, default='0d', doc=''):
+        """Retrieve a value as a datetime.timedelta."""
+        assert (
+            default is None or isinstance(default, str)
+        ), 'gettimespan("%s", %s) has an invalid default value type.' % (key, repr(default))
+        value = self._get(key, default=default, doc=doc, type_hint='timedelta')
+        if value is None:
+            return None
+
+        conversion = {'d': 'days', 'h': 'hours', 'm': 'minutes', 's': 'seconds'}
+        value, unit = value[:-1], value[-1:]
+        try:
+            value = float(value)
+            unit = conversion[unit]
+        except ValueError:
+            logger.exception("Unable to cast %s as float for the key %s.", repr(value), key)
+            raise
+        except KeyError:
```

Comment:
`KeyError` should not leak out to the caller, it's an implementation detail here. Given the other methods, it should probably be converted to a `ValueError`.